roles/aces-marketplace/tasks/main.yml

```diff
@@ -1,21 +1,91 @@
 ---
-- name: ensure apt cache is up to date
+- name: "ensure apt cache is up to date"
   apt: update_cache=yes
-  
-- name: ensure packages are installed
+
+# Add java repository from PPA and install.
+- apt_repository:
+    repo: ppa:webupd8team/java
+# Frontend repos
+- apt_repository:
+    repo: ppa:certbot/certbot
+
+- name: "ensure packages are installed"
   apt: name={{item}}
   with_items:
-    - postgresql
-    - postgresql-contrib
-    - python-psycopg2
+    - ufw
+    - oracle-java8-installer
+    - maven
+    - pwgen
     - nginx
+    - python-software-properties
+    #?? curl -sL https://deb.nodesource.com/setup_7.x | sudo -E bash -
+    - nodejs
+    # Frontend dependencies
+    - software-properties-common
+    - certbot
+
+- name: Configure ufw rules
+  ufw: rule={{ item.rule }} port={{ item.port }} proto={{ item.proto }}
+  with_items:
+    - { rule: 'allow', port: '22', proto: 'tcp' }
+    - { rule: 'allow', port: '80', proto: 'tcp' }
+    - { rule: 'allow', port: '443', proto: 'tcp' }
+
+- name: "Create database"
+  import_role:
+    name: database
+  vars:
+    database_host: localhost
+    database_name: default
+    database_user: default_user
+    database_password: change_me
```

Comment:
these need to be variables passed in from the playbook, copy how it looks in the other roles